ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -217,6 +230,34 @@ public CancelHandle receiveInvocationRequest(@NotNull final InvocationRequest in
         return cancellationFlag::cancel;
     }
 
+    private static <A> boolean intersect(final Set<A> fst, final Set<A> snd) {
+        fst.retainAll(snd);
+        return !fst.isEmpty();
+    }
+
+    private static Set<String> typesOf(final Object o) {
```

Comment:
You can load classes from modules without having an explicit dependency, if it comes to that.  But I think investigating the individual problems is a better course of action.

Any idea why we haven't run into these problems in earlier versions?